Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=d6b54841f4ddd836c886d1e6ac381cf309ee98a3
Commit:     d6b54841f4ddd836c886d1e6ac381cf309ee98a3
Parent:     4fb6de2561d5c47f3539a7b412108164d79fbb96
Author:     Evgeniy Dushistov <dushistov@mail.ru>
AuthorDate: Tue Jan 6 14:42:38 2009 -0800
Committer:  Linus Torvalds <torvalds@linux-foundation.org>
CommitDate: Tue Jan 6 15:59:27 2009 -0800

    minix: fix add link's wrong position calculation
    
    Fix the add link method.  The oosition in the directory was calculated in
    wrong way - it had the incorrect shift direction.
    
    [akpm@linux-foundation.org: coding-style fixes]
    Signed-off-by: Evgeniy Dushistov <dushistov@mail.ru>
    Cc: Nick Piggin <nickpiggin@yahoo.com.au>
    Cc: <stable@kernel.org>		[2.6.lots]
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

 fs/minix/dir.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/fs/minix/dir.c b/fs/minix/dir.c
index f704338..d4946c4 100644
--- a/fs/minix/dir.c
+++ b/fs/minix/dir.c
@@ -280,7 +280,7 @@ int minix_add_link(struct dentry *dentry, struct inode *inode)
 	return -EINVAL;
 
 got_it:
-	pos = (page->index >> PAGE_CACHE_SHIFT) + p - (char*)page_address(page);
+	pos = page_offset(page) + p - (char *)page_address(page);
 	err = __minix_write_begin(NULL, page->mapping, pos, sbi->s_dirsize,
 					AOP_FLAG_UNINTERRUPTIBLE, &page, NULL);
 	if (err)
